src/OrbitQt/CaptureOptionsDialog.cpp

```diff
@@ -194,6 +224,45 @@ DynamicInstrumentationMethod CaptureOptionsDialog::GetDynamicInstrumentationMeth
   ORBIT_UNREACHABLE();
 }
 
+void CaptureOptionsDialog::SetPureOrNotTracepoint(bool is_pure) {
+  ui_->tracepointCallstackCollectionCheckBox->setChecked(!is_pure);
+}
+
+bool CaptureOptionsDialog::GetPureOrNotTracepoint() const {
+  return !ui_->tracepointCallstackCollectionCheckBox->isChecked();
+}
+
+void CaptureOptionsDialog::SetTracepointCallstackMethod(
```

Comment:
Nit: From a decoupling perspective I would suggest to make some name changes. The capture options dialog doesn't know that we use trace points for the call stack collection. It only knows that we offer the user to collect call stacks when a thread state change. And this happens to use Linux trace points. But it's not for the dialog to decide that we use trace points.

Imagine we were adding support for Windows in the future. Ideally we want to use the same option in the dialog, but the collection mechanism won't be tracepoints but ETW events.

(This is only a nit about naming. Feel free to ignore)

Reply:
Dpme.